### ISSUE TITLE
Doesn't Seem to Handle Certain Characters

### ISSUE THREAD

**Issue by MicahLyle** [dhruvmanila/remove-print-statements#31]
Here's an example run from `pre-commit` (Python 3.11, v0.5.0 of this library)

```
remove print statements..................................................Failed
- hook id: remove-print-statements
- exit code: 123

Could not read file 'some_projects/some_folder/tests/conftest.py', skipping: 'charmap' codec can't decode byte 0x9d in position 5935: character maps to <undefined>
some_file.py
  43 print(cp_str)
  53 print(f'({attribute}) "{original_found}" -> ' f'"{replace}"')
  67 print(f'((inner) {attribute}) "{original_found}" -> ' f'"{replace}"')
  73 print(f"({attribute}) {initial_string} -> " f"{new_string}")
  112 print(f"{src_path} : Transforming HTML.")

1 file would be transformed, 5 print statements would be removed, 1 file would fail to transform
```

I think this is the problematic line in the file:

```
assert f"“{expected_comment}”" in body
```

Notice the fancy quote characters, (`“`, etc.)

My proposed fix would be:

```
    try:
        with open(filename, encoding="utf-8") as f:
            code = f.read()
    except Exception as exc:  # pragma: no cover
        click.secho(f"Could not read file {filename!r}, skipping: {exc}", fg="red")
        report.failure_count += 1
        return
```

^ And then doing the same thing with the `open` for the `write`.

I'm not 100% positive about this, but I'm pretty sure that Python files are encoded in "utf-8" (correct me if I'm wrong), so that should work.

**Comment by dhruvmanila** [comment]
Thank you for a detailed issue :)

> I think this is the problematic line in the file:
> 
> ```
> assert f"“{expected_comment}”" in body
> ```
> 
> Notice the fancy quote characters, (`“`, etc.)

I'm unable to reproduce using the above code. I've tried running using `pre-commit` as well but it still passed.

Can you give me the output of the following code? This will basically return the platform dependent encoding used by default. Also, what platform are you on?

```python
import locale

print(locale.getencoding())
```

Regarding the error message, you could just read the text around that location to actually get the character where it fails. This is basically moving the file pointer to 5 characters before the point where the error occurs (5935) and reading 10 bytes.

```python
with open('some_projects/some_folder/tests/conftest.py') as fd:
    fd.seek(5930)
	print(fd.read(10))
```

**Comment by MicahLyle** [comment]
```
In [1]: import locale

In [2]: print(locale.getencoding())
cp1252
```

I reproduced it by cloning the codebase writing some tests (and checking against a temp file I made).

This may be due to my default encoding (probably is).

With the PR (https://github.com/dhruvmanila/remove-print-statements/pull/32), if I remove the `encoding=` statements (two places), the tests fail (see below)

```
λ pytest . --sw
======================================================= test session starts =======================================================
platform win32 -- Python 3.11.2, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\micah\core\proj\temp\remove-print-statements, configfile: pyproject.toml
plugins: cov-3.0.0
collected 41 items
stepwise: no previously failed tests, not skipping.

tests\test_e2e.py ..F

============================================================ FAILURES =============================================================
___________________________________________ test_verbose_output_with_unicode_characters ___________________________________________

tmp_path = WindowsPath('C:/Users/micah/AppData/Local/Temp/pytest-of-micah/pytest-133/test_verbose_output_with_unico0')

    def test_verbose_output_with_unicode_characters(tmp_path: str) -> None:
        runner = CliRunner()
        newlines = 10
        with runner.isolated_filesystem(tmp_path) as temp_dir:
            Path(temp_dir).joinpath("hello_unicode.py").write_text(
                'print("“hello”")' + "\n" * newlines + 'print("“world”")', encoding="utf-8"
            )
            result = runner.invoke(main, ["hello_unicode.py", "--verbose"])

>       assert result.exit_code == 0
E       assert 123 == 0
E        +  where 123 = <Result SystemExit(123)>.exit_code

tests\test_e2e.py:41: AssertionError
```

If I put them back in (as I did in the PR) the tests all pass.

**Comment by MicahLyle** [comment]
Also see https://github.com/dhruvmanila/remove-print-statements/pull/32#issuecomment-1481682598

**Comment by MicahLyle** [comment]
@dhruvmanila if you do want to merge the PR (totally understand if there are edge cases I didn't think about), I just realized I forgot to bump the version. Maybe `v0.6.0` now since this can be considered a potentially breaking change?

**Comment by dhruvmanila** [comment]
closed by #32
